test/graph_test/base_graph_test.cpp

```diff
@@ -96,7 +96,23 @@ void BaseGraphTest::createDBAndConn() {
 }
 
 void BaseGraphTest::initGraph(const std::string& datasetDir) const {
-    if (conn) { // normal conn
+    // if we are importing from a previous kuzu version
+    // such as dataset/tmp/0.9.0_prev_exported_dbs/tinysnb
+    if (TestHelper::E2E_IMPORT_TEST_DIR != "dataset/") {
+        // falls out of the remaining conditionals
+        // i.e if conn is null we need to use the connMap
+        // as per the conn documentation
+        Connection* connection = conn ? conn.get() : (connMap.begin()->second).get();
+        // see migration related documentation
+        std::string query = "IMPORT DATABASE '" + datasetDir + "';";
+        std::cout << "Starting to execute query: " << query << std::endl;
+        auto result = connection->query(query);
+        std::cout << "Executed query: " << query << std::endl;
+        if (!result->isSuccess()) {
+            throw Exception(stringFormat("Failed to execute statement: {}.\nError: {}", query,
+                result->getErrorMessage()));
+        }
+    } else if (conn) { // normal conn
         TestHelper::executeScript(datasetDir + TestHelper::SCHEMA_FILE_NAME, *conn);
         TestHelper::executeScript(datasetDir + TestHelper::COPY_FILE_NAME, *conn);
     } else {
```

Comment:
While you're at it, an nice optional change here would be to check if datasetDir is "empty" and just skip the load. Don't have to do in this PR though.